ext/imap/php_imap.c

```diff
@@ -3714,7 +3714,7 @@ PHP_FUNCTION(imap_mail_compose)
 				convert_to_string_ex(pvalue);
 				bod->md5 = cpystr(Z_STRVAL_P(pvalue));
 			}
-		} else if (Z_TYPE_P(data) == IS_ARRAY) {
+		} else if (Z_TYPE_P(data) == IS_ARRAY && topbod->type == TYPEMULTIPART) {
```

Comment:
Should we warn in this case?

Reply:
This is basically the reverse of the "cannot generate multipart e-mail without components" error below, so might make sense to have symmetric treatment. I don't particularly care though.

Reply:
So far there is no warning, and the memory leak is relatively harmless. So maybe only warn for PHP 8.0+?

Reply:
Eh, it's ext/imap, probably not worth the bother...

Reply:
Well, we should do something about that. :)